### PR TITLE
DROOLS-4138 Building a KJAR using Decision Table Templates throws NPE

### PR DESCRIPTION
**Pull request by carlmes**
Added the missing class properties into the fromProperties() and toProperties() methods of the DecisionTableConfiguration.java object.

- Red Hat support case: https://access.redhat.com/support/cases/#/case/02397413
- JIRA bug report: https://issues.jboss.org/browse/DROOLS-4138

The updated .properties file that is generated after this fix now looks similar to this example:
```
#Configuration properties for resource: ExamplePolicyPricingTemplateData.xls
#Thu Jun 06 09:54:44 EDT 2019
kie.resource.conf.class=org.drools.core.builder.conf.impl.DecisionTableConfigurationImpl
resource.type=xls
drools.dt.type=XLS
drools.dt.trimCell=false
drools.dt.template.file0=src/main/resources/decision-table-template/BasePricing.drt
drools.dt.template.row0=3
drools.dt.template.col0=3
drools.dt.template.file1=src/main/resources/decision-table-template/PromotionalPricing.drt
drools.dt.template.row1=18
drools.dt.template.col1=3

```

### CODE REVIEW

**Comment by ghost** [comment]
Can one of the admins verify this PR? Comment with 'ok to test' to start the build.

**Comment by carlmes** [comment]
- Red Hat support case: https://access.redhat.com/support/cases/#/case/02397413
- JIRA bug report: https://issues.jboss.org/browse/DROOLS-4138

File: drools-core/src/main/java/org/drools/core/builder/conf/impl/DecisionTableConfigurationImpl.java [hunk 1]
```diff
@@ -89,13 +95,34 @@ public Properties toProperties() {
...
+            String file = prop.getProperty( DROOLS_DT_TEMPLATE_FILE_PREFIX + i, null );
+            if ( file == null ) {
+                break;
+            }
+            int row = Integer.parseInt( prop.getProperty( DROOLS_DT_TEMPLATE_ROW_PREFIX + i ) );
+            int col = Integer.parseInt( prop.getProperty( DROOLS_DT_TEMPLATE_COL_PREFIX + i ) );
+            Resource resource = ResourceFactory.newFileResource( file ).setSourcePath( file );
+            addRuleTemplateConfiguration( resource, row, col );
+            i++;
+        } while ( i < DROOLS_DT_TEMPLATE_MAX_TEMPLATES );
```
Comment (carlmes) [inline drools-core/src/main/java/org/drools/core/builder/conf/impl/DecisionTableConfigurationImpl.java:125]: I'm not thrilled about this method of looping through key/value pairs in order to find the last row. Perhaps there's a better way to implement the do > while loop?

**Comment by carlmes** [comment]
TODO - should we update the unit tests for this class to make sure property generation methods are successful?

**Comment by mariofusco** [comment]
See explanation in https://issues.jboss.org/browse/DROOLS-4138

**Pull request closed without merging**